@xen-orchestra/immutable-backups/doc.md

```diff
@@ -0,0 +1,147 @@
+- Watch all the `xo-vm-backups/<vm uuid>/vdis` sub folders for new job id
+- Watch all the `xo-vm-backups/<vm uuid>/vdis/<job uuid>/` sub folders for new job vdis
+- Watch all the `xo-vm-backups/<vm uuid>/vdis/<job uuid>/<vdi uuid>` sub folders for new vhd and alias.vhd file, ignoring dot files
+- Watch all the `xo-vm-backups/<vm uuid>/vdis/<job uuid>/<vdi uuid>/data` (only for increme)
+
+Make any new file in `xo-vm-backups/<vm uuid>/vdis/<job uuid>/<vdi uuid>` immutable
+
+#### vhd files
+
+the vhd files are uploaded to a dot file, and renamed on success. They are already handled by the watcher.
+
+#### vhd blocks
+
+- when a new folder is created in `xo-vm-backups/<vm uuid>/vdis/<job uuid>/<vdi uuid>/data`
+  - register a watcher of this folder `xo-vm-backups/<vm uuid>/vdis/<job uuid>/<vdi uuid>/data/<uuid>`
+  - wait for the bat/footer and header, then make the folder immutable recursively
+  - stop watching `xo-vm-backups/<vm uuid>/vdis/<job uuid>/<vdi uuid>/data/<uuid>`
+- alias are already handled by the watcher
+
+## Releasing protection on old enough files on a remote
+
+the `lift` process will list the files old enough in the `.immutability` folder and apply the individual lifting process
+
+This process should be run at least once a day.
+
+## Troubleshooting
+
+### some files are still locked
+
+launch the `lift` cli with `--no-cache`. This will do a fill filesystem traversal, and can be resource heavy, but will unlock all file based on their creation date.
+
+### make remote fully mutable again
+
+- Update the immutability setting with a 0 duration
+- launch the `lift` cli.
+- remove the `protect` service
+
+### increasing the immutability duration
+
+this will prolong immutable file, but won't protect files that are already out of immutability
+
+### reducing the immutability duration
+
+change the setting, and launch the `lift` cli , or wait for next planed execution
+
+### why my incremental backup are not marked as protected in XO ?
+
+to protect an incremental backup, all the chain must be under protection . To ensure at least 7 days of backups are protected, you need to set the immutability duration and retention at 14 days, the full backup interval at 7 days
+
+That means that if the last backup chain is cmplete ( 7 backup ) it is completly under protection, and if not, the precedent chain is also under protection. K are key backups, and d are delta
+
+```
+Kd Kdddddd Kdddddd K #  8 backups protected, 2 chains
+K Kdddddd Kdddddd Kd #  9 backups protected, 2 chains
+ Kdddddd Kdddddd Kdd # 10 backups protected, 2 chains
+ Kddddd Kdddddd Kddd # 11 backups protected, 2 chains
+ Kdddd Kdddddd Kdddd # 12 backups protected, 2 chains
+ Kddd Kdddddd Kddddd # 13 backups protected, 2 chains
+ Kdd Kdddddd Kdddddd #  7 backups protected, 1 chain since precedent full is now mutable
+Kd Kdddddd Kdddddd K #  8 backups protected, 2 chains
+```
+
+### Why does the protect process don't to start
```

Comment:
This sentence should probably be rephrased like : "Why doesn't the protect process start ?"